### PR TITLE
Added Advance tools to engineering cyborg

### PR DESCRIPTION
**Pull request by JayBaez**
What is in this PR:
Added the cyborg ATMOS holofan into the techweb for researching. You can research and print the upgrade board for the engineering cyborg. Added a new techweb research node for advance tools. The advance tools are the experimental tools (jaws of life, hand drill, tricorder, NOT experimental welding tool) and the Bluespace RPED. 

Purpose of this PR:
The engineering cyborg already has tools for humans, as the Geiger counter or analyzer, the ATMOS is too OP to have at the start so it can be researched. 
The BSRPED and advance tools give the borg a useful upgrade as the round advances.

:cl:  
rscadd: Added advance tools/ BSRPED/ ATMOS holofan to engineering cyborg
/:cl:

### CODE REVIEW

**Comment by JamieD1** [comment]
Jamie helped

**Comment by JayBaez** [comment]
ignore the various and redundant commits. I committed mistakes and was too lazy to make a new branch to remove it.

**Comment by Jarod1200** [comment]
borg powercreep moment

**Comment by UselessTheremin** [comment]
you really sat down, looked at the engiborg, and thought to yourself... "this needs a buff"

lmao

**Comment by JayBaez** [comment]
> you really sat down, looked at the engiborg, and thought to yourself... "this needs a buff"
> 
> lmao

It's just a quality of life

**Comment by Jarod1200** [comment]
Holofan and BRPED are too op for borgs period. There are literally 0 drawbacks to the engi borg with this PR.

**Comment by JayBaez** [comment]
> Holofan and BRPED are too op for borgs period. There are literally 0 drawbacks to the engi borg with this PR.

What would be OP would be giving the engineering cyborg the experimental welder. The holofan goes with the whole "keep human alive" tools like the Geiger counter and analyzer. The BSRPED is later game research where most things would have already been upgraded.

**Comment by Jarod1200** [comment]
The atmos holofan is a weapon of station ending destruction. "Whoops I just released my fusion cloud into medbay after welding the firelocks open as malf ai"

**Comment by adamsong** [comment]
> Whoops I just released my fusion cloud into medbay after welding the firelocks open as malf ai

Which of those steps used a holofan?

**Comment by ShatteredParadox** [comment]
> > Whoops I just released my fusion cloud into medbay after welding the firelocks open as malf ai
> 
> Which of those steps used a holofan?

Holding back the toxic/intensely hot gas, and at a push of a button, releasing the hot deadly gas through all of the open firelocks. Basically being able to remotely trigger the attack.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -747,7 +747,168 @@
...
+			R.module.remove_module(CB, TRUE)
+
+		for(var/obj/item/wirecutters/cyborg/WC in R.module.modules) 
+			R.module.remove_module(WC, TRUE)
+
+		for(var/obj/item/multitool/cyborg/MT in R.module.modules) 
+			R.module.remove_module(MT, TRUE)
+
+		for(var/obj/item/t_scanner/TS in R.module.modules) 
+			R.module.remove_module(TS, TRUE)
```
Comment (JamieD1) [inline code/game/objects/items/robot/robot_upgrades.dm]: T ray needed
```suggestion
```

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -747,7 +747,168 @@
...
+
+		var/obj/item/multitool/cyborg/MT = locate() in R.module.modules
+		MT = new(R.module)
+		R.module.basic_modules += MT
+		R.module.add_module(MT, FALSE, TRUE)
+
+		var/obj/item/t_scanner/TS = locate() in R.module.modules
+		TS = new(R.module)
+		R.module.basic_modules += TS
+		R.module.add_module(TS, FALSE, TRUE)
```
Comment (JamieD1) [inline code/game/objects/items/robot/robot_upgrades.dm]:
```suggestion
```

**Review by JamieD1 — APPROVED** [review]
No experimental t ray

**Comment by JayBaez** [comment]
You play a hard game Jamie. I’m all good with your recommendation

**Comment by JamieD1** [comment]
> You play a hard game Jamie. I’m all good with your recommendation

Its more the fact you remove the t ray and dont give a replacement with the advanced tools so you are basically deleting the t ray

**Review by JamieD1 — CHANGES_REQUESTED** [review]
Remove the BSRPED from its own node and unlock it on the normal BSRPED to prevent wasting of resources if "noone will ever use it"

Also disable the option for disabling the barriers remotely.

Shout if you need help with any of these.

**Comment by JayBaez** [comment]
> > You play a hard game Jamie. I’m all good with your recommendation
> 
> Its more the fact you remove the t ray and dont give a replacement with the advanced tools so you are basically deleting the t ray

But doesn't the Tricorder have a T-ray scanner built in? along with power, atmos?

**Comment by Jarod1200** [comment]
Tricorder does not have a t ray

**Comment by JayBaez** [comment]
> Tricorder does not have a t ray

Then I must be monkey brained

**Comment by JayBaez** [comment]
> Remove the BSRPED from its own node and unlock it on the normal BSRPED to prevent wasting of resources if "noone will ever use it"
> 
> Also disable the option for disabling the barriers remotely.
> 
> Shout if you need help with any of these.

prob will cause my time to work on this has dropped since school started back up again.

File: code/modules/research/techweb/layout.dm [hunk 3]
```diff
@@ -509,4 +512,4 @@
 
 /datum/techweb_node/mech_xray
         ui_x = -640
-        ui_y = -896
\ No newline at end of file
+        ui_y = -896
```
Comment (JamieD1) [inline code/modules/research/techweb/layout.dm]:
```suggestion
        ui_y = -896
```

**Review by patpol4 — CHANGES_REQUESTED** [review]
What is the point of the map change here? Looks like you're just doing changes from #12270 ?

**Comment by patpol4** [comment]
Also I would like to state i think this pr is bad and i don't like it

**Comment by SomeguyManperson** [comment]
image
wtf happened here

**Comment by patpol4** [comment]
@JayBaez is anything happening to this

**Comment by Bibby0110** [comment]
Fix merge conflicts + whatever jamie did.
Good luck with your school :)

**Comment by JayBaez** [comment]
> @JayBaez is anything happening to this

yea

**Pull request closed without merging**